### PR TITLE
Limit the number of db reads using duet.Limiter

### PR DESCRIPTION
**Pull request by senecameeks**
Related to b/376053385

### CODE REVIEW

File: cirq-core/cirq/work/sampler.py [hunk 1]
```diff
@@ -311,32 +291,16 @@ async def run_batch_async(
...
-            await duet.sleep(1)  # Delay for 1 second between chunk
-            results.extend(
-                await duet.pstarmap_async(
-                    self.run_sweep_async, zip(program_chunk, params_chunk, reps_chunk)
-                )
-            )
-
-        return results
+        return await duet.pstarmap_async(
+            self.run_sweep_async, zip(programs, params_list, repetitions, [limiter] * len(programs))
```
Comment (NoureldinYosri) [inline cirq-core/cirq/work/sampler.py:302]: shouldn't this be `repetitions), limit=...)`?

https://github.com/google/duet/blob/6817af592ecdf2aac0565a0de62bd7287fc9203a/duet/api.py#L171-L173

Comment (senecameeks) [reply cirq-core/cirq/work/sampler.py:302]: That's for write requests... the intention is to pass a Limiter object to jobs.results_async so we can limit read requests from the spanner database

**Review by NoureldinYosri — APPROVED** [review]
LGTM%nit

File: cirq-google/cirq_google/engine/processor_sampler.py [hunk 2]
```diff
@@ -70,6 +76,11 @@ async def run_sweep_async(
             snapshot_id=self._snapshot_id,
             device_config_name=self._device_config_name,
         )
+
+        if isinstance(job, EngineJob):
+            print("calling...")
```
[the remark this answers is not in the document]
Comment (senecameeks) [reply cirq-google/cirq_google/engine/processor_sampler.py]: Good catch, deleted

**Pull request merged**